packages/composites/augur/src/dataProviders/sportsdataio.ts

```diff
@@ -119,14 +217,155 @@ export const create: Execute = async (input) => {
+  Status: string
+}
+
+interface Fighter {
+  Active: boolean
+  FighterId: number
+  Winner: boolean
+}
+
+const getLeagues = async (id: string, sport: string, exec: Execute): Promise<string[]> => {
+  const input = {
+    id,
+    data: {
+      sport,
+      endpoint: 'leagues'
+    }
+  }
+  const response = await exec(input)
+  return (response.result as { Key: string }[]).map(res => res.Key)
+}
+
+const getFightSchedule = async (id: string, sport: string, league: string, season: string, exec: Execute): Promise<FightSchedule[]> => {
+  const input = {
+    id,
+    data: {
+      sport,
+      league,
+      season,
+      endpoint: 'schedule'
+    }
+  }
+  const response = await exec(input)
+  return (response.result as FightSchedule[])
+    .filter(event => event.Active)
+}
+
+const getFights = async (id: string, sport: string, eventId: number, exec: Execute): Promise<Fight[]> => {
+  const input = {
+    id,
+    data: {
+      sport,
+      eventId,
+      endpoint: 'event'
+    }
+  }
+  const response = await exec(input)
+  const fights = (response.result as FightEvent).Fights
+  return fights.filter(fight => fight.Active)
+}
+
+export const createFighter: Execute = async (input) => {
+  const validator = new Validator(input, createParams)
+  if (validator.error) throw validator.error
+
+  const sport = validator.validated.data.sport.toLowerCase()
+  if (!SPORTS_SUPPORTED.includes(sport)) {
+    throw Error(`Unknown sport for Sportsdataio: ${sport}`)
+  }
+
+  const daysInAdvance = validator.validated.data.daysInAdvance
+  // const contract: ethers.Contract = validator.validated.data.contract
+
+  const sportsdataioExec = Sportsdataio.makeExecute(Sportsdataio.makeConfig(Sportsdataio.NAME))
+
+  const fights: Fight[] = []
+
+  const leagues = await getLeagues(input.id, sport, sportsdataioExec)
+  for (const league of leagues) {
+    const schedule = (await getFightSchedule(input.id, sport, league, `${new Date().getFullYear()}`, sportsdataioExec))
+      .filter(event => event.Status === "Scheduled")
+
+    for (const event of schedule) {
+      const eventFights = (await getFights(input.id, sport, event.EventId, sportsdataioExec))
+        .filter(fight => fight.Status === "Scheduled")
+        .map(fight => ({ ...fight, DateTime: event.DateTime }))
+      fights.push(...eventFights)
+    }
+  }
+
+  Logger.debug(`Augur sportsdataio: Got ${fights.length} fights from data provider`)
+  let skipNullDate = 0, skipDaysInAdvance = 0, skipOddNumberFighters = 0, cantCreate = 0
+
+  // filter markets and build payloads for market creation
+  const createEvents: CreateFighterEvent[] = []
+  for (const fight of fights) {
+    if (!fight.DateTime) {
+      skipNullDate++
+      continue
+    }
+    const startTime = Date.parse(fight.DateTime)
+    const diffTime = startTime - Date.now()
+    if (diffTime / (1000 * 3600 * 24) > daysInAdvance) {
+      skipDaysInAdvance++
+      continue
+    }
+
+    // TODO: How to check if fight already exists?
```

Comment:
Use the `events` mapping on the MMALinkMarketFactory contract, here: 
https://github.com/AugurProject/turbo/blob/dev/packages/smart/contracts/turbo/MMALinkMarketFactory.sol#L60

if eventStatus field is 0 then the market does not already exist.

It will look something like:
```typescript
if((await contract.events(eventId)).eventStatus === 0) {
    cantCreate++
    continue
}